compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -479,10 +479,13 @@ impl<'a> Resolver<'a> {
                     ),
                 );
                 err.help(
-                    &format!("it is currently only allowed to use either `{0}` or `{{ {0} }}` as generic constants", name)
+                    &format!("note: type parameters are currently not permitted in anonymous constants")
                 );
```

Comment:
Using both 'help' and 'note: ' is confusing, try `err.note("type parameters ...")` instead maybe.